### ISSUE TITLE
Problem installing clean astroquery

### ISSUE THREAD

**Issue by tomdonaldson** [astropy/astroquery#1920]
I'm getting an error trying to install a clean version of astroquery.  Hopefully it's a transient issue and will resolve itself  shortly, but if not here is what I did:
```
conda create --quiet --yes --name try_astroquery python=3.8
conda activate try_astroquery
pip install --pre astroquery
```

The error is:
```
ERROR: Requested astroquery from 
https://files.pythonhosted.org/packages/65/41/3cc633331a276f78aa5fa110c7dc71e146dd7d14346f3460534a264e0a5d
/astroquery-0.4.2.dev6449.tar.gz#sha256=df7786a5ba87c966f705f6fce02d73bd7c48a748d7e0a89f0531bcd544a7a074
has different version in metadata: '0.4.2.dev0'
```

**Comment by pllim** [comment]
I see that `astroquery` is still using `astropy-helpers` to generate the version:

https://github.com/astropy/astroquery/blob/fc70b94d13fa44789fe69e133bc264eff872d97a/setup.py#L68-L69

This issue might be naturally solved by applying APE 17 (#1668).

**Comment by bsipocz** [comment]
I don't see this error, and can't recall anyone else running into it. What is your pip version? (I'm not on the latest one, so would guess this may be a side effect of the new resolver).

**Comment by pllim** [comment]
For me, no error with `pip 20.2.4` but same error with `pip 20.3.1`. (`pip` was installed with `conda` in Python 3.8 env.)

**Comment by bsipocz** [comment]
oh, great, I see a new 🔥 on the horizon. 

What was still on my todo list is to tag 0.4.2, so I'll get to it hopefully during the weekend, and I believe that would be the quickest workaround (the CD is broken at the moment anyway, so this issue would not pop up for the near future anyway).

Thanks @tomdonaldson for reporting this

**Comment by drtobybrown** [comment]
I'm getting the same error with pip 20.3.1. Works using pip 20.2.4

**Comment by ceb8** [comment]
@tomdonaldson and I discovered another workaround to this issue: 
`pip install --pre astroquery --use-deprecated=legacy-resolver`

**Comment by bsipocz** [comment]
This is now resolved with the new 0.4.2 release. A proper solution will come with the infra refactoring, including changes to how we generate dev version numbers.
